src/t_stream.c

```diff
@@ -3608,6 +3985,95 @@ void xdelCommand(client *c) {
     if (ids != static_ids) zfree(ids);
 }
 
+/* Used by xdelexCommand() */
+typedef enum XDelexRes {
+    XDELEX_NO_ID = -2,           /* ID not found in the stream. */
+    XDELEX_DELETED = 0,          /* Message deleted. */
+    XDELEX_STILL_REFERENCED = 1, /* Message not deleted (still referenced). */
+} XDelexRes;
+
+/* XDELEX <key> [KEEPREF|DELREF|ACKED] [IDS <numids> <id ...>]
+ *
+ * Removes specified entries from the stream. Returns an array of status codes for
+ * each ID, indicating whether it was deleted, still referenced, or not found. */
+void xdelexCommand(client *c) {
+    kvobj *kv = lookupKeyWriteOrReply(c, c->argv[1], shared.czero); 
+    if (kv == NULL || checkType(c, kv, OBJ_STREAM)) return;
+    stream *s = kv->ptr;
+
+    /* Parse command options */
+    streamAckDelArgs args;
+    if (!streamParseAckDelArgsOrReply(c, 2, &args)) return;
+
+    /* We need to sanity check the IDs passed to start. Even if not
+     * a big issue, it is not great that the command is only partially
+     * executed because at some point an invalid ID is parsed. */
+    streamID static_ids[STREAMID_STATIC_VECTOR_LEN];
+    streamID *ids = static_ids;
+    if (args.numids > STREAMID_STATIC_VECTOR_LEN)
+        ids = zmalloc(sizeof(streamID)*args.numids);
+    for (int j = 0; j < args.numids; j++) {
+        if (streamParseStrictIDOrReply(c,c->argv[j+args.startidx],&ids[j],0,NULL) != C_OK) goto cleanup;
+    }
+
+    int first_entry = 0;
+    int deleted = 0;
+    addReplyArrayLen(c, args.numids);
+    for (int j = 0; j < args.numids; j++) {
+        int res = XDELEX_NO_ID;
```

Comment:
This default value always gets overwritten below at `res = can_delete ? XDELEX_DELETED : XDELEX_STILL_REFERENCED;`